server/portal/asgi.py

```diff
@@ -5,8 +5,12 @@
 
 import os
 import django
-from channels.routing import get_default_application
+from channels.routing import get_default_application, ProtocolTypeRouter
+from django.core.asgi import get_asgi_application
+
 
 os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'portal.settings.settings')
 django.setup()
-application = get_default_application()
+application = ProtocolTypeRouter({
+    "http": get_asgi_application(),
+})
```

Comment:
Related: https://channels.readthedocs.io/en/stable/releases/3.0.0.html#deprecations

Reply:
Do websocket notifications work without defining a mapping for `"websocket` in the `ProtocolTypeRouter?

Reply:
The websocket mapping is defined in [routing.py](https://github.com/TACC/Core-Portal/blob/task/WP-271--django-upgrade/server/portal/routing.py) and from my testing notifications are working fine